app/precompiled.py

```diff
@@ -234,9 +225,55 @@ def overlay_template_areas(src_pdf, page_number=None, overlay=True):
     return png_from_pdf(pdf, page_number)
 
 
-def get_invalid_pages(src_pdf):
-    pdf_to_validate = _overlay_printable_areas(src_pdf)
-    return list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+def get_invalid_pages_with_message(src_pdf):
+    message = ""
+    invalid_pages = []
+    are_pages_A4_portrait_oriented, invalid_pages = _are_pages_A4_portrait_oriented(src_pdf)
+    if not are_pages_A4_portrait_oriented:
+        message = 'The page orientation is landscape instead of portrait on '
+    else:
+        pdf_to_validate = _overlay_printable_areas(src_pdf)
+        invalid_pages = list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+        if len(invalid_pages) > 0:
+            message = 'Content in this PDF is outside the printable area on '
+    if len(invalid_pages) > 0:
+        message += unescaped_formatted_list(
+            invalid_pages,
+            before_each='',
+            after_each='',
+            prefix='page',
+            prefix_plural='pages'
+        )
+    return (invalid_pages, message)
+
+
+def _are_pages_A4_portrait_oriented(src_pdf):
```

Comment:
we can change this to `get_pages_with_invalid_orientation_or_size`, and then rather than returning a tuple, just return the list of invalid page numbers.

then we can just check if the list is empty to see whether it was okay or not

Reply:
hmm true